src/azure-cli/azure/cli/command_modules/vm/azure_stack/_actions.py

```diff
@@ -47,84 +47,83 @@ def _get_thread_count():
 
 def load_images_thru_services(cli_ctx, publisher, offer, sku, location, edge_zone, architecture):
     from concurrent.futures import ThreadPoolExecutor, as_completed
+    VMImage = import_aaz_by_profile(cli_ctx.cloud.profile, "vm.image")
+
+    VMImageListPublishers = VMImage.ListPublishers
+    VMImageListOffers = VMImage.ListOffers
+    VMImageListSkus = VMImage.ListSkus
+    VMImageList = VMImage.List
 
     all_images = []
-    client = _compute_client_factory(cli_ctx)
     if location is None:
         location = get_one_of_subscription_locations(cli_ctx)
 
     def _load_images_from_publisher(publisher):
         from azure.core.exceptions import ResourceNotFoundError
         try:
-            if edge_zone is not None:
-                offers = edge_zone_client.list_offers(location=location, edge_zone=edge_zone, publisher_name=publisher)
-            else:
-                offers = client.virtual_machine_images.list_offers(location=location, publisher_name=publisher)
+            offers = VMImageListOffers(cli_ctx=cli_ctx)(command_args={
+                'location': location,
+                'publisher': publisher,
+            })
```

Comment:
The Azure Stack doesn't need to support `edge_zone`, right?

Reply:
yes, you are right